### ISSUE TITLE
Player's UID is not unique any more

### ISSUE THREAD

**Issue by shenzhoukai** [cheahjs/palworld-steam-id-to-player-uid#6]
SteamID                               UID (useSteam)      UID (noSteam)
76561199186857738           1009231501           3262899357
76561200516214127           243748559             3262899357

Although the second player is not valid in Steam any more but the algo of SteamID to UID (noSteam) will generate duplicated UID.
Will it cause game account stealing problem if those players join the same server ?

**Comment by shenzhoukai** [comment]
76561200300123504 and 76561199961080696 has the same UID (useSteam)

**Comment by shenzhoukai** [comment]
76561199734837953 has the same UID (useSteam) 335190009 with the following Steam Player's UID (noSteam)

76561201226335938
76561199163620927
76561198338269899
76561199604003633

Warning! 76561199163620927, 76561198338269899 and 76561199604003633 are all valid users in Steam.

**Comment by shenzhoukai** [comment]
SteamID UID (useSteam) UID (noSteam)
76561198168093340 3889347819 3346717094
76561198315537556 3889347819 3346717094

These two players have all duplicated UID both in useSteam and noSteam mode.

**Comment by shenzhoukai** [comment]
SteamID UID (useSteam) UID (noSteam)
76561199337100197 3859255899 1115524924
76561199429105744 3859255899 1115524924
76561198946530363 3859255899 1115524924

76561199337100197 and 76561199429105744 are valid Steam player.

**Comment by cheahjs** [comment]
> Although the second player is not valid in Steam any more but the algo of SteamID to UID (noSteam) will generate duplicated UID.

Yes, this is expected as the algorithm takes a value with 32-bits of information (Steam ID), and hashes it into a 32-bit value (UID), which will cause collisions.

With how algorithm works (Steam ID hashes into a Steam UID hashes into a NoSteam UID), if players share the same Steam UID, they will share the same NoSteam UID.

> Will it cause game account stealing problem if those players join the same server ?

If they end up joining the same server, yes, they will use the same character save data.

**Comment by shenzhoukai** [comment]
Are you the developer of this game? Or can you contact the dev team of PalWorld to report this issue?

**Comment by thejcpalma** [comment]
@cheahjs is there a way to chat with you for 10 minutes about this project? Specially the database lookup and how you implemented it on your address. I wanted to have a database too but force bruting all steam ids seems excessive
